### PR TITLE
Added multicursor improvements

### PR DESCRIPTION
**Pull request by jmbockhorst**
* Added Ctrl+Shift+D to select next occurence of selection #5322
* Left and right cursor movement and enter/return now works for multiple cursors

multicursordemo

### CODE REVIEW

File: PowerEditor/src/Parameters.cpp [hunk 1]
```diff
@@ -141,6 +141,12 @@ static const WinMenuKeyDefinition winKeyDefs[] =
 	{ VK_NULL,    IDM_EDIT_SORTLINES_DECIMALCOMMA_DESCENDING,   false, false, false, nullptr },
 	{ VK_NULL,    IDM_EDIT_SORTLINES_DECIMALDOT_ASCENDING,      false, false, false, nullptr },
 	{ VK_NULL,    IDM_EDIT_SORTLINES_DECIMALDOT_DESCENDING,     false, false, false, nullptr },
+	{ VK_D,		  IDM_EDIT_SELECTNEXTOCCURENCE,				    true,  false, true,  nullptr },
```
Comment (Faq) [inline PowerEditor/src/Parameters.cpp]: used tab instead of 4 spaces?

Comment (jmbockhorst) [reply PowerEditor/src/Parameters.cpp]: Yes, I will fix that.

File: PowerEditor/src/NppCommands.cpp [hunk 2]
```diff
@@ -1570,6 +1570,99 @@ void Notepad_plus::command(int id)
...
+
+			// Convert char to TCHAR
+			WcharMbcsConvertor *wmc = WcharMbcsConvertor::getInstance();
+			UINT cp = static_cast<UINT>(_pEditView->execute(SCI_GETCODEPAGE));
+			const TCHAR* textToFind = wmc->char2wchar(selectedText, cp);
+
+			// Find position of next occurence
+			int posFound = _pEditView->searchInTarget(textToFind, textLen, currentPos + 1, docLength);
+
+			if (posFound > currentPos) {
```
Comment (donho) [inline PowerEditor/src/NppCommands.cpp]: bad coding style:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

File: PowerEditor/src/NppCommands.cpp [hunk 3]
```diff
@@ -1570,6 +1570,99 @@ void Notepad_plus::command(int id)
...
+				int start = posFound + textLen;
+				int end = posFound;
+
+				// Detect if selection happened the other way
+				if (currentPos != selectonEnd) {
```
Comment (donho) [inline PowerEditor/src/NppCommands.cpp]: bad coding style:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

File: PowerEditor/src/NppCommands.cpp [hunk 4]
```diff
@@ -1570,6 +1570,99 @@ void Notepad_plus::command(int id)
...
+		case IDM_EDIT_MULTINEWLINE:
+		{
+			// Get the number of selection
+			int numSelections = int(_pEditView->execute(SCI_GETSELECTIONS));
+
+			// Keep an array of the new positions
+			std::vector<int> selectionPositions;
+
+			// Loop through each selection
+			for (int i = 0; i < numSelections; i++) {
```
Comment (donho) [inline PowerEditor/src/NppCommands.cpp]: bad coding style:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

File: PowerEditor/src/NppCommands.cpp [hunk 5]
```diff
@@ -1570,6 +1570,99 @@ void Notepad_plus::command(int id)
...
+
+				// Set the current selection to main
+				_pEditView->execute(SCI_SETMAINSELECTION, i);
+
+				// Execute the new line command on the selection
+				_pEditView->execute(SCI_NEWLINE);
+			}
+
+			// Add all of the cursors back where they should be
+			for (int i = 0; i < ((signed int)(selectionPositions.size()) - 1); i++) {
```
Comment (donho) [inline PowerEditor/src/NppCommands.cpp]: bad coding style:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

**Comment by jmbockhorst** [comment]
Sorry, I realized about the coding style after I committed. I will fix it.

File: PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp [hunk 6]
```diff
@@ -2589,6 +2589,91 @@ void ScintillaEditView::currentLinesDown() const
 	execute(SCI_SCROLLRANGE, execute(SCI_GETSELECTIONEND), execute(SCI_GETSELECTIONSTART));
 }
 
+void ScintillaEditView::multiCursorLeftOrRight(int direction, bool shouldExt) const
+{
+    // Get the number of selection
+    int numSelections = int(execute(SCI_GETSELECTIONS));
+
+    // Use default functions when there is one selection
+    if (numSelections == 1) {
```
Comment (donho) [inline PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp]: bad coding style in the added codes:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

**Comment by jmbockhorst** [comment]
The release x64 build failed but the console log was empty, so I'm not sure what happened. It built fine locally.

**Comment by donho** [comment]
@jmbockhorst  Don't worry about Appveyor, it does build on my PC too.

**Pull request closed without merging**